ang/civicase/CaseListTable.js

```diff
@@ -134,23 +134,27 @@
       $scope.$bindToRoute({expr: 'page.num', param: 'cpn', format: 'int', default: 1});
     }
 
-    function caseIsFocusedWatchHandler () {
-      $timeout(function () {
-        var $actHeader = $('.act-feed-panel .panel-header');
-        var $actControls = $('.act-feed-panel .act-list-controls');
+    /**
+     * Fetch additional information about the contacts
+     *
+     * @param {array} caseObj
+     */
+    function fetchContactsData (event, cases) {
```

Comment:
missing param doc, and wrong param name on the doc.

Reply:
Please refactor the function as follows:

```js
function fetchContactsData (event, cases) {
  var contactIds = cases.map(function (caseObj) {
    return getAllContactIdsForCase(caseObj);
  });

  contactIds = _.chain(contactIds).flatten().unique().value();

  ContactsDataService.add(contactIds);
}

function getAllContactIdsForCase (caseObj) {
  var contactIds = _.map(caseObj.contacts, 'contact_id');
  var activitiesContactIds = _.map(caseObj.allActivities, function (activity) {
    return _.chain(activity)
      .pick(['assignee_contact_id', 'target_contact_id', 'source_contact_id'])
      .values()
      .value();
  });

  return _.concat(contactIds, activitiesContactIds);
}
```

Reply:
Both fixed. 
PS: the exact code did not work, but I have divided my previous function as requested.